api/v1alpha4/clusterclass_webhook.go

```diff
@@ -132,6 +161,107 @@ func (in *ClusterClass) validate(old *ClusterClass) error {
 			)
 		}
 	}
+	return allErrs
+}
+
+func (in ClusterClass) validateRefTemplates() field.ErrorList {
+	var allErrs field.ErrorList
+
+	if !strings.HasSuffix(strings.ToLower(in.Spec.Infrastructure.Ref.Kind), "template") {
```

Comment:
Also, should we check that the value is something + Suffix (or Len(value)> Len(suffix))?

Reply:
@fabriziopandini I wasn't sure if the suffix being capitalized is a requirement. Looks like it is. I can revert it back to checking for capitals. Also thanks for the pointer for the constant. :) 

> Also, should we check that the value is something + Suffix (or Len(value)> Len(suffix))?

+1

Reply:
> FYI, we already have a const defined in https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/external/util.go#L36

Just realized that we cannot use this constant from here as it will lead to a cyclic dependency between `controller/external` and `api/v1alpha4` packages. 
Instead we can move the constant to the `api/v1alpha4` package and use it in `controller/external`, wdyt?

Reply:
> A little copying is better than a little dependency.
> https://go-proverbs.github.io/ 
:)